### DISCUSSION TITLE
How can I disable the symlink icon?

### DISCUSSION THREAD

**Discussion by Bonen0209** [nvim-tree/nvim-tree.lua#1678] [Q&A]
### Description

Here is the minimum config of reproducing the errors.

```
nvim_tree.setup {
  renderer = {
    icons = {
      show = {
        file = false,
        folder = false,
        folder_arrow = false,
        git = false
      }
    }
  }
}
```

After using this config, symlink's icons still show up like the picture down below.

![Screen Shot 2022-10-12 at 18 12 05](https://user-images.githubusercontent.com/32999887/195316004-39eb5298-02a0-4ae8-a8f4-90cbe5bf66af.png)

Although I can disable symlink's icons manually by adding additional lines to the config, as shown below, it will be nice to have an option to disable symlink's icons.

```
nvim_tree.setup {
  renderer = {
    icons = {
      show = {
        file = false,
        folder = false,
        folder_arrow = false,
        git = false
      },
      glyphs = {
        symlink = ""
      }
    }
  }
}
```

### Neovim version

```text
NVIM v0.8.0
Build type: Release
LuaJIT 2.1.0-beta3
```

### Operating system and version

macOS 12.6

### nvim-tree version

b01e7be

### Minimal config

```lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvt-min/site]]
local package_root = "/tmp/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      "nvim-tree/nvim-tree.lua",
      "nvim-tree/nvim-web-devicons",
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing nvim-tree and dependencies."
  vim.fn.system { "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
vim.opt.termguicolors = true
vim.opt.cursorline = true

-- MODIFY NVIM-TREE SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
_G.setup = function()
  require("nvim-tree").setup {
    renderer = {
      icons = {
        show = {
          file = false,
          folder = false,
          folder_arrow = false,
          git = false
        }
      }
    }
  }
end
```

### Steps to reproduce

nvim -nu /tmp/nvt-min.lua

### Expected behavior

![Screen Shot 2022-10-12 at 18 27 54](https://user-images.githubusercontent.com/32999887/195319507-7255266d-38df-4503-899d-923e62c19a13.png)

### Actual behavior

![Screen Shot 2022-10-12 at 18 12 05](https://user-images.githubusercontent.com/32999887/195316004-39eb5298-02a0-4ae8-a8f4-90cbe5bf66af.png)

**Comment by alex-courtis** [comment]
The symlink icon can be disabled by setting `nvim-tree.renderer.icons.symlink_arrow` to space or empty, however I don't think that's the solution you are after.

You can disable symlink source/arrow/target completely via `nvim-tree.renderer.symlink_destination`.

**Reply by trmckay** [reply]
I think this is a bug. The following config still seems to render icons on symlinks for me.

```
{
    renderer = {
        icons = {
            show = {
                file = false,
                folder = false,
                folder_arrow = false,
                git = false,
            },
            symlink_arrow = "",
        },
        symlink_destination = false,
    },
}
```

**Comment by alex-courtis** [comment]
I see: the icon is still showing.

This should meet your requirements:
```lua
  require("nvim-tree").setup({
    renderer = {
      icons = {
        show = {
          file = false,
          folder = false,
          folder_arrow = false,
          git = false,
        },
        glyphs = {
          symlink = "",
        },
      },
      symlink_destination = false,
    },
  })
```

**Comment by Bonen0209** [comment]
Hi @alex-courtis,

I am @Bonen0209. First of all, thanks for answering my question.

The problem I try to discuss is that could we offer an option to turn off symlink icon like `renderer.icons.show.symlink = false` ?

I will try my best to see if I can contribute this options to nvim-tree.

**Reply by alex-courtis** [reply]
> The problem I try to discuss is that could we offer an option to turn off symlink icon like `renderer.icons.show.symlink = false` ?

That would be fantastic. It is a much nicer solution and would be gratefully appreciated.

Perhaps something like:
```lua
      show = {
        file = true,
        folder = true,
        folder_arrow = true,
        git = true,
        symlink_file = false,
        symlink_folder = false,
      },
```

Please see https://github.com/nvim-tree/nvim-tree.lua/blob/master/CONTRIBUTING.md
